a10_octavia/a10_config.py

```diff
@@ -77,6 +81,48 @@ def __init__(self, config_dir=None, config=None, provider=None):
         self._config.octavia_conf_dir = '/etc/octavia/'
         self._load_config()
 
+    def get_rack_dict(self):
+        rack_dict = {}
+        if self._conf.has_section("RACK_VTHUNDER") and self._conf.has_option("RACK_VTHUNDER", "devices"):
+            project_conf = self._conf.get('RACK_VTHUNDER', 'devices')
+            rack_conf = ast.literal_eval(project_conf.strip('"'))
+            validation_flag = False
+            try:
+                for i in range(len(rack_conf["device_list"])):
+                    project_id = rack_conf["device_list"][i]["project_id"]
+                    ip_address = rack_conf["device_list"][i]["ip_address"]
+                    undercloud = bool(rack_conf["device_list"][i]["undercloud"])
+                    username = rack_conf["device_list"][i]["username"]
+                    password = rack_conf["device_list"][i]["password"]
+                    device_name = rack_conf["device_list"][i]["device_name"]
+                    axapi_version = rack_conf["device_list"][i]["axapi_version"]
+                    role = rack_conf["device_list"][i]["role"]
+                    topology = rack_conf["device_list"][i]["topology"]
+                    validation_flag = self.validate(project_id, ip_address, username,
+                                                    password, axapi_version,
+                                                    undercloud, device_name,
+                                                    role, topology)
+
+                    if validation_flag:
+                        vthunder_conf = data_models.VThunder(project_id=project_id,
+                                                             ip_address=ip_address,
+                                                             undercloud=undercloud,
```

Comment:
@Hunter, I was looking at your review comment of unpacking dictionary, But I feel, we should make sure that user have created a valid dictionary entry with required parameters. So, I feel this code is necessary to validate the parameters before creating a valid vThunder object

Reply:
It's simply not the right way to do this check. There is no point in wasting about 80 bytes for variables that'll be declared and used only once and then promptly garbage collected leading to increased overhead.

The validation function already takes in these parameters. So, there is no reason to ensure that they exist in this way when we already have a check ensuring that they exist.

The biggest issue though is that the data model allows for these to not be specified. If they are required, then they need to be positional arguments not keyword args.

Reply:
If you want better error handling, then make a custom exception for this module that'll catch the TypeError when all the positional args aren't filled or a wrong keyword is specified. You can then have it state a message that "X" wasn't specified in the config.

Reply:
Also, the issue in the loop still stands of how you're iterating over it. The pythonic way to iterate is with `for-each` loops not `for` loops.

Reply:
Still working on the custom error and data model management, will update